### PR TITLE
feat(volume group): extend specs api to include volume group specs

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
1. Add volume group specs to be a part of specs api
2. Add test to validate volume group loading in memory on startup.

### CODE REVIEW

File: control-plane/agents/src/bin/core/tests/volume/volume_group.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                    volume_group: item.0.map(|val| VolumeGroup::new(val.to_string())),
+                    ..Default::default()
+                },
+                None,
+            )
+            .await;
+    }
+
+    // Restart the core-agent.
+    let _ = cluster.composer().restart("core").await;
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/tests/volume/volume_group.rs]:
```suggestion
    cluster.restart_core().await;
```
Ensures we clean up etcd lock

File: control-plane/agents/src/bin/core/tests/volume/volume_group.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    // Wait for the core-agent grpc to come up.
+    let _ = cluster.node_service_liveness(None).await;
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/tests/volume/volume_group.rs]:
```suggestion
    // Wait for the core-agent grpc to come up.
    cluster
        .node_service_liveness(None)
        .await
        .expect("Should have restarted by now");
```

File: control-plane/grpc/src/operations/registry/traits.rs [hunk 3]
```diff
@@ -102,6 +107,14 @@ impl TryFrom<registry::Specs> for transport::Specs {
                 }
                 replica_specs
             },
+            volume_groups: {
+                let mut vg_specs: Vec<VolumeGroupSpec> =
+                    Vec::with_capacity(value.volume_groups.len());
+                for vg_spec in value.volume_groups {
+                    vg_specs.push(VolumeGroupSpec::try_from(vg_spec)?);
+                }
+                vg_specs
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/registry/traits.rs]:
```suggestion
value.volume_groups.into_iter().map(VolumeGroupSpec::try_from).collect::<Result<Vec<VolumeGroupSpec>, SvcError>>()?
```

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/agents/src/bin/core/tests/volume/volume_group.rs (hunk 2 above)
Comment (Abhinandan-Purkait) [reply control-plane/agents/src/bin/core/tests/volume/volume_group.rs]: Done.

File: control-plane/agents/src/bin/core/tests/volume/volume_group.rs (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/agents/src/bin/core/tests/volume/volume_group.rs]: Cool.

File: control-plane/grpc/src/operations/registry/traits.rs (hunk 3 above)
Comment (Abhinandan-Purkait) [reply control-plane/grpc/src/operations/registry/traits.rs]: Good point. Changed it for the other resources as well.

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**